Galactic Radio Telescope

hexylena opened the pull request:
Howdy all,

I'd like to re-introduce this project:
![](https://github.com/erasche/galactic-radio-telescope/raw/master/media/grt-small.png)

The Galactic Radio Telescope (GRT) has had some progress today, and it's gotten to a point where it can be used to start collection of data, while looking towards analysing that data quite a bit further down the line.
## What?

The GRT aims to collect job run information from Galaxy servers that opt into the process.

Galaxy Admins can sign in to the GRT, register their Galaxy servers, and get back API keys per-Galaxy instance. Then they set up a cron job (preferably weekly, but we may explore options here later), and submit job data to the GRT.
## Why?

https://oc.hx42.org/grt/ the about page should cover most questions.
## GRT Repo

https://github.com/erasche/galactic-radio-telescope
## Demo Instance

I've deployed a demo instance here:
https://oc.hx42.org/grt/

And I've posted all of my org's Galaxy's data to it.

![utvalg_837](https://cloud.githubusercontent.com/assets/458683/15794669/49314fe8-29dc-11e6-83e1-b3f2f314840f.png)

(Warning, accessing things like the "CPT Phage Galaxy" information page in the Galaxy list will be SLOW. It's running off of sqlite, with a large number of uncached queries. If this PR gets accepted, I'll optimize it a lot)

hexylena committed:
Add Galactic Radio Telescope script

martenson commented:
It is very nice Eric! I see great potential. :koala:

hexylena committed:
Linting

hexylena committed:
Make time length an option

hexylena committed:
Submit more metadata

hexylena committed:
Document GRT

hexylena committed:
E303

hexylena committed:
Ignore metadata setting

hexylena committed:
Change metric captured

ericenns committed:
Added in last_job_id_sent and removed days options so no duplicates will be sent.

ericenns committed:
Removed call to args.days

ericenns committed:
Added in tool_blacklist

ericenns committed:
Moved almost all cmd line parameters to config file.

ericenns committed:
Added grt_url to grt.ini.sample and removed default from argparse

ericenns committed:
Flake8 fixes

ericenns committed:
Fixed remaining flake8 issues

natefoo commented:
I want this.

afgane commented:
Potential platform for analyzing the collected data: https://www.tacc.utexas.edu/research-development/tacc-projects/karnak

hexylena commented:
After talking to people, this has been deemed a good idea / useful. 

As such I'll be rewriting parts (the whole) of the backend to be a bit more efficient and cleanly architected.

That said, the script portion of it is mostly done (minus sending all tool params) which we'll add shortly

ericenns committed:
Added in params reporting with initial sanitization

hexylena commented on scripts/grt.py:
```diff
@@ -7,6 +7,8 @@
 import argparse
 import sqlalchemy as sa
 import yaml
+import copy
```
unused import

ericenns committed:
Removed unused copy import

ericenns committed:
Added upload1 to tool blacklist

jmchilton committed:
grt.ini.sample is a YAML file, git it a YAML extension.

jmchilton committed:
Give grt a top-level docstring.

jmchilton committed:
Define a main() function for grt client.

This is purely stylistic - but I prefer to structure code to allow testing even if there are no actual test cases yet.

For more information see discussion here - http://stackoverflow.com/questions/4041238/why-use-def-main.

jmchilton committed:
Fix some obvious Python 2isms in grt client...

... and one instance of bare exception.

jmchilton committed:
Remove custom Galaxy config parsing in grt.py.

Use existing abstractions - so things like environment variable based configuration will work.

jmchilton committed:
Refactor grt.py method names to be clear public interface is only main().

jmchilton commented:
:+1: if changes in https://github.com/erasche/galaxy/pull/16 are acceptable and accepted. Thanks for working on this @erasche - this is a great start!

hexylena committed:
Merge pull request #16 from jmchilton/grt

Touch up grt.py.

hexylena commented:
Thanks @ericenns, awesome work :) It's definitely a start @jmchilton, hopefully we can start from here and improve over time. I was really waiting on this to be accepted before putting more time into the backend, so if you're +1ing it... :)
